plugins/woocommerce/src/Admin/API/Reports/DataStore.php

```diff
@@ -1391,8 +1396,20 @@ protected function get_attribute_subqueries( $query_args ) {
 					$sql_clauses['join'][] = "JOIN {$wpdb->prefix}woocommerce_order_itemmeta as {$join_alias} ON {$join_alias}.order_item_id = {$table_to_join_on}.order_item_id";
 				}
 
-				// phpcs:ignore WordPress.DB.PreparedSQL.InterpolatedNotPrepared
-				$sql_clauses['where'][] = $wpdb->prepare( "( {$join_alias}.meta_key = %s AND {$join_alias}.meta_value {$comparator} %s )", $meta_key, $meta_value );
+				$in_comparator = '=' === $comparator ? 'in' : 'not in';
+
+				// Add subquery for products ordered using attributes not used in variations.
+				// phpcs:disable WordPress.DB.PreparedSQL.InterpolatedNotPrepared
+				$sql_clauses['where'][] = $wpdb->prepare(
+					"
+					( ( {$join_alias}.meta_key = %s AND {$join_alias}.meta_value {$comparator} %s ) or (
+						{$wpdb->prefix}wc_order_product_lookup.variation_id = 0 and {$wpdb->prefix}wc_order_product_lookup.product_id {$in_comparator} ( select product_id from {$wpdb->prefix}wc_product_attributes_lookup where is_variation_attribute=0 and term_id = %s )
```

Comment:
Added an order sub query, to see if a simple product has a matching attribute id.
Note: this query works only for simple products with global attributes, given the attributes filter only shows global attributes to filter by.

Reply:
I'm not a fan of this line. Maybe extracting the subquery to a new variable would improve its legibility.

Maybe something like this would work:
```
$prepared_term_id = intval( $term_id );
$subquery = "SELECT product_id FROM {$wpdb->prefix}wc_product_attributes_lookup WHERE is_variation_attribute = 0 AND term_id = {$prepared_term_id}";
$where_condition = "
	( ( {$join_alias}.meta_key = %s AND {$join_alias}.meta_value {$comparator} %s ) OR
		({$wpdb->prefix}wc_order_product_lookup.variation_id = 0 AND
		{$wpdb->prefix}wc_order_product_lookup.product_id {$in_comparator} ({$subquery}) ) )";
$sql_clauses['where'][] = $wpdb->prepare( $where_condition, $meta_key, $meta_value );
```

Reply:
Addressed as part of: https://github.com/woocommerce/woocommerce/pull/44901/commits/96f2b45d1048adc5c6f41db57bb768cc3b8be124